### ISSUE TITLE
Importing modules by name is case sensitive

### ISSUE THREAD

**Issue by jdhitsolutions** [PowerShell/PowerShell#1621]
## Steps to reproduce

Start a PowerShell session.
Remove-Module PSReadline
Import-Module PSReadline
## Expected behavior

Expect to be able to re-import PSReadline. This works with other modules.
## Actual behavior

Error reporting module can't be found.

<img width="362" alt="psreadlineerror" src="https://cloud.githubusercontent.com/assets/12089920/17369373/17b028fe-5966-11e6-9c62-f8ef8cefec58.png">
## Environment data

``` powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      5.1.10032.0
PSEdition                      PowerShellCore
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   3.0.0.0
GitCommitId                    v6.0.0-alpha.7
CLRVersion
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by jpsnover** [comment]
I believe that as a general principle, PowerShell should be as case-preserving/case insensitive as possible while not breaking the underlying OS.

So in the case of the Unix File system manipulation and environmental variables - these directly affect the underlying OS and integration with other tools so we have to be case sensitive.

In the case of PowerShell variable names - there is no interaction between them and the underlying OS so we should be case insensitive.

So then issue with Module Names is that they are a tweener.  On the one hand there is a strong mapping to an underlying OS - a foldername.  On the other hand, theyy refer to a pure PowerShell concept - a PowerShell Module.   So you could argue it either way.

To me, the thing that pushes this decisively to the Case Insensitive solution is the effect the issue has on Ecosystem sharing.  If module names are case sensitive in Unix, then Tom might publish Module TEST and Mary might publish Module test.  But then if a Windows users tries to use both these modules, they are going to collide and they fail.   

I'm going to propose an acid test we need to apply:  Does case sensitivity break sharing artifacts between Windows and Linux?    If the answer is yes, then the prima facia answer should be that we should be case insensitive.

Ergo - Module names should be case insensitive. 

Does that sound right?

**Comment by ForNeVeR** [comment]
Guys, please note that it isn't about "Windows is case-sensitive and Linux is not". That's "most common Windows file systems are case-insensitive" and "most common Linux file systems are case-sensitive". And, AFAIK, macOS is case-insensitive too. You could switch this option either on Windows, on Linux or on macOS.

That decision (at least about the file names) shouldn't be tied to platform, but to underlying FS (and every OS could at once mount multiple file systems with different settings).

**Comment by Zsoldier** [comment]
Noticing that $env variables are case-sensitive as well.  $env:PSMODULEPATH does not equal $env:PSModulePath

**Comment by andschwa** [comment]
Environment variables on Linux are case sensitive.

**Comment by joeyaiello** [comment]
This needs to get fixed along with #706. Adding it to the usability project.

**Comment by joeyaiello** [comment]
Based on @jpsnover's reply and the logic that the @PowerShell/powershell-committee landed on in #706, we should close on the fact that importing two modules with the same case-insensitive names should have the same behavior as whatever happens when you import two modules with the same name from different paths (either an error, overwriting the existing one in memory, or silently doing nothing).
